### PR TITLE
Added panic handler middleware; returning correct errors in `/api/days/brief`

### LINKED ISSUE #9: `days/brief` api method doesn't satisfy a design requirement
The (draft) API documentation requires all the methods except `/login` to always return a json object with the `ok` key, however, `/api/days/brief` does not satisfy that requirement

### PR DESCRIPTION
**Pull request by kolayne**
This PR introduces middleware, recovering from panic and returning `500 Internal Server Error` (with a template file for UI and the format described in the docs for API) and fixes #9

### CODE REVIEW

File: src/common.go [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
 
-import "log"
-
 //Panics if `err` is not `nil`
 //
 // @param err Error to be checked for equality to `nil`
 func panicIfError(err error) {
 	if err != nil {
-		log.Panic(err)
+		panic(err)
```
Comment (kolayne) [inline src/common.go:8]: Moved logging to middleware, because we can start panicking without calling `panicIfError`

File: src/middleware.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+func UIPanicHandlerMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				log.Println(err)
+
+				w.WriteHeader(http.StatusInternalServerError)
+				if err := tmpl.ExecuteTemplate(w, "500", fmt.Sprint(err)); err != nil {
+					log.Println(err)
+					_, _ = w.Write([]byte("It failed :("))
```
Comment (kolayne) [inline src/middleware.go:41]: I think there is no need to handle the error of `w.Write`, because what can we do if there is an error?

**Review by DarkSquirrelComes — APPROVED** [review]
Ok, but I think in the future we shouldn't return all error's content to users.

**Review by VladimirKoryakin — APPROVED** [review]
Nice

**Review by gccenin — APPROVED** [review]

**Pull request merged**